src/components/queryBuilder/TimeField.tsx

```diff
@@ -3,19 +3,20 @@ import { SelectableValue } from '@grafana/data';
 import { InlineFormLabel, Select } from '@grafana/ui';
 import { selectors } from './../../selectors';
 import { FullField } from 'types';
-import { isDateType } from './utils';
 
 interface TimeFieldEditorProps {
   fieldsList: FullField[];
   timeField: string;
   timeFieldType: string;
   onTimeFieldChange: (timeField: string, timeFieldType: string) => void;
+  timeFieldTypeCheckFn: (type: string) => boolean;
+  labelAndTooltip: typeof selectors.components.QueryEditor.QueryBuilder.TIME_FIELD;
```

Comment:
I made it a bit more configurable to reuse for Logs Volume time field selection, as we want to see only the `DateTime*` types there and specify different labels and tooltips.